src/analysis/inner_product_space/l2_space.lean

```diff
@@ -403,6 +404,33 @@ protected lemma tsum_inner_mul_inner (b : hilbert_basis ι 𝕜 E) (x y : E) :
   ∑' i, ⟪x, b i⟫ * ⟪b i, y⟫ = ⟪x, y⟫ :=
 (b.has_sum_inner_mul_inner x y).tsum_eq
 
+-- Note : this should be `b.repr` composed with an identification of `lp (λ i : ι, 𝕜) 2` with
+-- `pi_Lp 2 (λ i : ι, 𝕜)`, but we don't have this yet (July 2022).
+/-- A finite Hilbert basis is an orthonormal basis. -/
+protected def to_orthonormal_basis [fintype ι] (b : hilbert_basis ι 𝕜 E) :
+  orthonormal_basis ι 𝕜 E :=
+orthonormal_basis.mk b.orthonormal
+begin
+  rw [← set.image_univ, ← finset.coe_univ, ← finset.coe_image],
+  have := (span 𝕜 (finset.univ.image b : set E)).closed_of_finite_dimensional,
+  rw [← this.submodule_topological_closure_eq, finset.coe_image, finset.coe_univ, set.image_univ],
+  exact b.dense_span
+end
+
+@[simp] lemma coe_to_orthonormal_basis [fintype ι] (b : hilbert_basis ι 𝕜 E) :
+  (b.to_orthonormal_basis : ι → E) = b :=
+orthonormal_basis.coe_mk _ _
+
+protected lemma has_sum_orthogonal_projection {U : submodule 𝕜 E} [complete_space E]
+  [complete_space U] (b : hilbert_basis ι 𝕜 U) (x : E) :
+  has_sum (λ i, ⟪(b i : E), x⟫ • b i) (orthogonal_projection U x) :=
+begin
+  convert b.has_sum_repr (orthogonal_projection U x) using 2,
+  ext i,
+  rw [b.repr_apply_apply, coe_inner, ← inner_orthogonal_projection_left_eq_right,
+      orthogonal_projection_mem_subspace_eq_self]
```

Comment:
Can you please add this fact (`⟪b, orthogonal_projection U x⟫ = ⟪(b : E), x⟫`) as a simp-lemma and then make the proof `by simpa using ...`